### PR TITLE
[UC05#155#156] Added functionality for users to add own physical cards to one of theirs custom decks

### LINKED ISSUE #155: Contattare RPC inviando token, lista carta/e fisica/e, nome mazzo personalizzato [UC15]
_**Dod**_
- [ ]  La pull request ha un titolo e descrizione adeguata
- [ ]  La pull request è stata collegata con la issue del task
- [ ]  I requisiti funzionali del task collegato sono soddisfatti
- [ ]  Il codice si compila senza errori
- [ ]  Gli Unit Tests sono stati scritti e passati 
- [ ]  Gli Unit Tests esistenti sono stati passati
- [ ]  Il Code Review è stato fatto
- [ ]  La documentazione è stata aggiornata
- [ ]  La pull request è stata accettata e fatto merge sul branch development.

 **Effort effettivo:**

### LINKED ISSUE #156: Aggiornare la pagina dei mazzi con le carte fisiche aggiunte ai mazzi personalizzati [UC15]
_**Dod**_
- [ ]  La pull request ha un titolo e descrizione adeguata
- [ ]  La pull request è stata collegata con la issue del task
- [ ]  I requisiti funzionali del task collegato sono soddisfatti
- [ ]  Il codice si compila senza errori
- [ ]  Gli Unit Tests sono stati scritti e passati 
- [ ]  Gli Unit Tests esistenti sono stati passati
- [ ]  Il Code Review è stato fatto
- [ ]  La documentazione è stata aggiornata
- [ ]  La pull request è stata accettata e fatto merge sul branch development.

 **Effort effettivo:**

### PR DESCRIPTION
**Pull request by alessioarcara**
This PR implements #155, #156 and adds a new method called `addPhysicalCardsToCustomDeck` in the DecksActivity to add owned physical cards to a custom deck via RPC call. Additionally, the decks collection has been changed from a HashMap to a LinkedHashMap to ensure the ordering of keys (and thus deck names).
Tests for the new method were also added.

Oopsie Doopsie, we've reached 300 tests 🚀

### CODE REVIEW

**Review by dstops — APPROVED** [review]
Good Job, nothing to declare!

File: src/main/java/com/aadm/cardexchange/client/widgets/DeckWidget.java [hunk 1]
```diff
@@ -80,6 +82,7 @@ public DeckWidget(ImperativeHandleDeck deckHandler, ImperativeHandleCustomDeck c
     }
 
     public void setData(List<PhysicalCardWithName> data, String selectedCardId) {
+        Logger.getLogger("adsasd").info(data.toString());
```
Comment (alessiacrimaldi) [inline src/main/java/com/aadm/cardexchange/client/widgets/DeckWidget.java:85]: There is a logger to remove

File: src/main/java/com/aadm/cardexchange/client/widgets/DeckWidget.ui.xml [hunk 2]
```diff
@@ -40,9 +46,12 @@
...
     <g:HTMLPanel styleName="{style.deck}">
         <div class="{style.deckHeader}">
-            <h2 ui:field="deckName"/>
-            <g:HTMLPanel styleName="{style.actions}" ui:field="actions">
+            <div class="{style.deckNameWrapper}">
+                <h2 ui:field="deckName"/>
                 <g:Button styleName="deckButton {style.showButton}" ui:field="showButton">></g:Button>
+            </div>
+            <g:HTMLPanel styleName="{style.actions}" ui:field="actions">
+                <!--                <g:Button styleName="deckButton {style.showButton}" ui:field="showButton">></g:Button>-->
```
Comment (alessiacrimaldi) [inline src/main/java/com/aadm/cardexchange/client/widgets/DeckWidget.ui.xml:54]: Also here there's a comment to remove

**Review by alessiacrimaldi — APPROVED** [review]
Okay, all right! Just be careful to re-read the code when you’re making prints or comments which are no longer useful.

P.S: Oopsiee Doopsieee for the 300 tests 🥳

**Pull request merged**